chains/evm/contracts/onRamp/CommitOnRamp.sol

```diff
@@ -54,28 +55,34 @@ contract CommitOnRamp is Ownable2StepMsgSender, BaseOnRamp {
   /// @notice Forwards a message from CCV proxy to this verifier for processing and returns verifier-specific data.
   /// @dev This function is called by the CCV proxy to delegate message verification to this specific verifier.
   /// It performs critical validation to ensure message integrity and proper sequencing.
-  /// @param rawMessage The encoded message containing all necessary data for verification.
-  /// @param verifierIndex Index of this verifier in the message's verifier receipts array.
+  /// @param message Decoded MessageV1 payload for verification.
+  /// @param messageId keccak256(encodedMessage) of the message.
+  /// @param feeToken Fee token used for the message.
+  /// @param feeTokenAmount Amount of fee token provided.
+  /// @param verifierArgs Opaque verifier-specific args.
   /// @return Verifier-specific encoded data (nonce in case of commit onramp).
-  function forwardToVerifier(bytes calldata rawMessage, uint256 verifierIndex) external returns (bytes memory) {
-    Internal.EVM2AnyVerifierMessage memory message = abi.decode(rawMessage, (Internal.EVM2AnyVerifierMessage));
-
-    _assertNotCursed(message.header.destChainSelector);
-    _assertSenderIsAllowed(message.header.destChainSelector, message.sender);
+  function forwardToVerifier(
+    MessageV1Codec.MessageV1 calldata message,
+    bytes32 messageId,
+    address feeToken,
+    uint256 feeTokenAmount,
+    bytes calldata verifierArgs
+  ) external returns (bytes memory) {
+    (messageId); // silence unused; reserved for future use
+
+    _assertNotCursed(message.destChainSelector);
```

Comment:
I think we're fine to remove this as this contract doesn't really do anything special and CCVProxy already checks cursing